feat: support loacal tx history.

liuzjalex opened the pull request:
合并显示本地和 covalent 历史记录，目前还是以 covalent 的格式，后续开发交易队列管理的时候会一并修改。

@taimanhui 目前本地发出的的交易如果短时间内调用 getHistory 的话交易状态会判断成为 "dropped" ；可能是 getTransactionStatuses 的判断有问题？

liuzjalex committed:
feat: support loacal tx history.

liuzjalex committed:
fix: refresh pending tx bug.

taimanhui commented on packages/engine/src/proxy.ts:
```diff
@@ -676,19 +676,21 @@ class ProviderController extends BaseProviderController {
         status === TransactionStatus.INVALID
       ) {
-        ret[entryId] = HistoryEntryStatus.DROPPED;
+        if (Date.now() - createdAt > 60 * 5) {
```
move `Date.now()` out of the loop

liuzjalex committed:
fix: update gas spend.

liuzjalex committed:
fix: fix lint & improvement.

liuzjalex committed:
fix: use type guard instead.

liuzjalex committed:
fix: Merge commit 'ad1142a217b5c33cf24da8d3ed9e39a0095c2415' into merge_tx_history

* commit 'ad1142a217b5c33cf24da8d3ed9e39a0095c2415':
  OK-7903 Styles: Radio Button & i18n (#561)
  OK-7636: add the option to enable localAuthentication when setting password for the first time (#558)
  OK-6554, OK-7940, Added Pressed state & Haptics (#556)
  fix: balance read and type (#559)
  Token approve tx & max amount for NumberInput (#537)
  OK-7737: ios local credentials for eas build (#555)
  OK-7952: improve token list (#554)
  fix : search dapp state (#534)
  Feat: add discover history OK-7714 OK-7731 (#532)
  refactor: tokens redux/ useManageTokens hooks (#547)
  chore: update version v1.0.4 (#553)
  OK-7865: fix simulator on M1 (#552)
  refactor useToast hook (#551)
  OK-7633 (#549)
  OK-7749, OK-7920 fix recover account navigation. (#548)
